src/utils/format.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Uses Intl.NumberFormat style='decimal' to add thousands separator to a
+ * displayed number
+ *
+ * @param value
+ */
+export const formatNumber = (value: string | number | bigint | null) => {
+  const numValue = !value
+    ? 0
+    : typeof value === 'string'
+      ? parseFloat(value)
+      : typeof value === 'bigint'
+        ? Number(value)
+        : value;
+
+  return new Intl.NumberFormat(undefined, {
+    style: 'decimal',
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 4,
+  }).format(numValue);
+};
```

Comment:
Intl.NumberFormat `format` has supported BigInt since introduction of that primitive (cf. https://github.com/tc39/ecma402/pull/236 ) and basically does all this conversion work already, including `null` → `0`.
```suggestion
export const formatNumber = (value: string | number | bigint | null) => {
  return new Intl.NumberFormat(undefined, {
    style: 'decimal',
    minimumFractionDigits: 2,
    maximumFractionDigits: 4,
  }).format(value as any);
};
```
But really, `Intl.NumberFormat` is intended to be used in a "construct one, format many" pattern anyway:
```suggestion
export const formatNumber = new Intl.NumberFormat(undefined, {
  style: 'decimal',
  minimumFractionDigits: 2,
  maximumFractionDigits: 4,
}).format as (value: string | number | bigint | null) => string;
```

Reply:
Very informative feedback - thank you!